### ISSUE TITLE
False positive in detect-non-literal-regexp

### ISSUE THREAD

**Issue by Sjord** [returntocorp/sgrep-rules#1468]
[detect-non-literal-regexp](https://github.com/returntocorp/semgrep-rules/blob/develop/javascript/lang/security/audit/detect-non-literal-regexp.yaml)

corpus.js:
```
new RegExp(/a/)
```

Run semgrep:

```
semgrep --config 'r/javascript.lang.security.audit.detect-non-literal-regexp'
using config from https://semgrep.dev/r/javascript.lang.security.audit.detect-non-literal-regexp. Visit https://semgrep.dev/registry to see all public rules.
downloading config...
running 1 rules...
corpus.js
severity:warning rule:javascript.lang.security.audit.detect-non-literal-regexp.detect-non-literal-regexp: RegExp() called with a variable, this might allow an attacker to DOS your application with a long-running regular expression.
1:new RegExp(/a/)
ran 1 rules on 1 files: 1 findings
```
